src/main/java/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/machines/electric/geo/OilPump.java

```diff
@@ -97,25 +97,28 @@ protected void tick(Block b) {
 			}
 		}
 		else {
-			OreGenResource oil = OreGenSystem.getResource("Oil");
-			Chunk chunk = b.getChunk();
-			int supplies = OreGenSystem.getSupplies(oil, chunk, false);
-			
-			if (supplies > 0) {
-				for (int slot : getInputSlots()) {
-					if (SlimefunManager.isItemSimilar(inv.getItemInSlot(slot), new ItemStack(Material.BUCKET), true)) {
+			if (!inv.fits(SlimefunItems.BUCKET_OF_OIL, getOutputSlots())) {
+				return;
+			}
+			for (int slot : getInputSlots()) {
+				if (SlimefunManager.isItemSimilar(inv.getItemInSlot(slot), new ItemStack(Material.BUCKET), true)) {
+					OreGenResource oil = OreGenSystem.getResource("Oil");
+					Chunk chunk = b.getChunk();
+					int supplies = OreGenSystem.getSupplies(oil, chunk, false);
+					if(supplies > 0) {
 						MachineRecipe r = new MachineRecipe(26, new ItemStack[0], new ItemStack[] {SlimefunItems.BUCKET_OF_OIL});
-						
-						if (!inv.fits(SlimefunItems.BUCKET_OF_OIL, getOutputSlots())) {
-							return;
-						}
-						
+
 						inv.replaceExistingItem(slot, InvUtils.decreaseItem(inv.getItemInSlot(slot), 1));
 						processing.put(b, r);
 						progress.put(b, r.getTicks());
 						OreGenSystem.setSupplies(oil, chunk, supplies - 1);
 						break;
 					}
+					else {
+						ItemStack item = BlockStorage.getInventory(b).getItemInSlot(slot).clone();
+						BlockStorage.getInventory(b).replaceExistingItem(slot, null);
+						pushItems(b, item);
+					}
```

Comment:
This code still doesn't make much sense...
Why are you putting the bucket into the machine twice...?